tests/unit/contentproviders/test_ckan.py

```diff
@@ -0,0 +1,58 @@
+import os
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile, TemporaryDirectory
+
+import pytest
+
+from repo2docker.contentproviders import CKAN
+
+test_ckan = CKAN()
+test_hosts = [
+    (
+        [
+            "http://demo.ckan.org/dataset/sample-dataset-1",
+        ],
+        {
+            "dataset_id": "sample-dataset-1",
+            "api_url": "http://demo.ckan.org/api/3/action/",
+            "version": "1707387710",
+        },
+    )
+]
+
+
+@pytest.mark.parametrize("test_input, expected", test_hosts)
```

Comment:
This is currently failing with:

```
>       assert CKAN().detect(test_input[0]) == expected
E       AssertionError: assert {'api_url': '... '1709043532'} == {'api_url': '... '1707387710'}
E         Omitting 2 identical items, use -vv to show
E         Differing items:
E         {'version': '1709043532'} != {'version': '1707387710'}
E         Use -v to get more diff
```

Is there any way to lock this down some more?

Reply:
Oh, I forget that the demo site of CKAN will be rebuild occasionally. Instead I wrote a mock response for the `test_detect_ckan` test.